pkg/entrypoint/admissioncontroller/admissioncontroller.go

```diff
@@ -70,6 +72,14 @@ func newAdmissionController(ctx context.Context, log *logrus.Entry, configFile s
 		return nil, err
 	}
 
+	if cs.Properties.PrivateAPIServer {
+		hostname := os.Getenv("masterNodeName")
+		if hostname == "" {
+			return nil, fmt.Errorf("could not read hostname from env[masterNodeName]")
+		}
+		cs.Config.MasterKubeconfig.Clusters[0].Cluster.Server = "https://" + hostname
+	}
```

Comment:
I don't think that this contributes to the core issue, but I think it's worth fixing nonetheless.  Functionality roughly copied from the sync pod.